subprojects/docs/src/snippets/dependencyManagement/modelingFeatures-crossProjectPublications-advanced-published/kotlin/buildSrc/src/main/kotlin/com/acme/InstrumentedJarsPlugin.kt

```diff
@@ -83,4 +83,5 @@ class InstrumentedJarsPlugin @Inject constructor(
 
 }
 
-inline fun <reified T : Named> Project.namedAttribute(value: String) = objects.named(T::class.java, value)
+inline fun <reified T: Named> AttributeContainer.attribute(attr: Attribute<T>, value: String) =
+    attribute(attr, objects.named<T>(value))
```

Comment:
It might be nice to integrate this extension by default so users won't need to copy-paste it.

I can do that, however, for now, the PR is a mere documentation update

Reply:
Just in case, it looks like it requires `ObjectFactory`, so either each `AttributeContainer` implementation should have one injected, or an alternative option is to add a function to `abstract class ProjectDelegate : Project {
 ... inline fun <reified T: Named> AttributeContainer.attribute`. Then it would be available in the regular `kts` build files.